Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -424,6 +424,49 @@ namespace AzToolsFramework
             return newTemplateId;
         }
 
+        void PrefabSystemComponent::UpdateTemplateFilePath(TemplateId templateId, const AZ::IO::PathView& filePath)
+        {
+            auto findTemplateResult = FindTemplate(templateId);
+            if (!findTemplateResult.has_value())
+            {
+                AZ_Warning(
+                    "Prefab", false,
+                    "PrefabSystemComponent::UpdateTemplateFilePath - "
+                    "Template associated by given Id '%llu' doesn't exist in PrefabSystemComponent.",
+                    templateId);
+                return;
+            }
+
+            if (!filePath.IsRelative())
+            {
+                AZ_Warning(
+                    "Prefab", false,
+                    "PrefabSystemComponent::UpdateTemplateFilePath - "
+                    "Provided filePath must be relative.",
+                    templateId);
+                return;
+            }
+
+            Template& templateToChange = findTemplateResult->get();
+            if (templateToChange.GetFilePath() == filePath)
+            {
+                return;
+            }
+            
+            m_templateFilePathToIdMap.erase(templateToChange.GetFilePath());
+            m_templateFilePathToIdMap.emplace(AZStd::make_pair(filePath, templateId));
+
+            PrefabDom& prefabDom = templateToChange.GetPrefabDom();
+            PrefabDomValueReference pathReference = Prefab::PrefabDomUtils::FindPrefabDomValue(prefabDom, "Source");
+            if (pathReference)
+            {
+                const AZStd::string_view pathStr = filePath.Native();
+                pathReference->get().SetString(rapidjson::StringRef(pathStr.data(), pathStr.length()), prefabDom.GetAllocator());
```

Comment:
Should this be a `rapidjson::StringRef`? Using a `StringRef` will store a reference to a string instead of a copy, so the file path will need to outlive this particular RapidJSON value. If the file path isn't stored somewhere than the string will reference deleted memory.